nemo/collections/llm/gpt/model/mixtral.py

```diff
@@ -0,0 +1,174 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TYPE_CHECKING, Callable, Optional
+
+import torch
+import torch.nn.functional as F
+
+from nemo.collections.llm.gpt.model.base import GPTConfig, GPTModel
+from nemo.lightning import io, teardown
+
+if TYPE_CHECKING:
+    from transformers import MistralConfig, MistralForCausalLM
+
+    from nemo.collections.common.tokenizers.huggingface.auto_tokenizer import AutoTokenizer
+
+
+@dataclass
+class MixtralConfig(GPTConfig):
+    normalization: str = "RMSNorm"
+    activation_func: Callable = F.silu
+    position_embedding_type: str = "rope"
+    add_bias_linear: bool = False
+    gated_linear_unit: bool = True
+    apply_query_key_layer_scaling: bool = False  # TODO: Should this be True?
+
+    num_layers: int = 32
+    hidden_size: int = 4096
+    num_attention_heads: int = 32
+    num_query_groups: int = 8
+    ffn_hidden_size: int = 14336
+    max_position_embeddings: int = 4096  # 32768
+    seq_length: int = 4096  # 32768
+    # MoE
+    num_moe_experts: int = 8
+    moe_router_topk: int = 1
+
+    init_method_std: float = 0.02
+    layernorm_epsilon: float = 1e-5
+    # rotary
+    rotary_percent: float = 0.5
+    rotary_base: float = 10000
+
+
+class MixtralModel(GPTModel):
+    def __init__(self, config: Optional[MixtralConfig] = None, optim_config=None, tokenizer=None):
+        _tokenizer = tokenizer or HFMixtralImporter('mistralai/Mixtral-8x7B-v0.1').tokenizer
```

Comment:
We updated this in the `MistralModel`: https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/llm/gpt/model/mistral_7b.py#L44-L50